### ISSUE TITLE
Random Charecters added to URL

### ISSUE THREAD

**Issue by craftzneko** [grafana/grafana#87820]
### What happened?

I had a previously working dashboard that had a Data link. Now characters are added to values. Data link looks like this

https://logic.azure.com/workflows/invoke?api-version=2016-06-01&GroupID=%EF%BB%BF%EF%BB%BF854032434532345344%EF%BB%BF

The group ID has this appended to it %EF%BB%BF%EF%BB%BF8540 and after it %EF%BB%BF

The data is fetch from a sql table and has no zero width charecters i have replaced the data in there to be sure. The data link looks like this

https://logic.azure.com/workflows/invoke?api-version=2016-06-01&GroupID=﻿﻿${__data.fields.AzureGroupID}﻿

### What did you expect to happen?

GroupID is inserted without random string charecters applied

### Did this work before?

Yes, this latest version

### How do we reproduce it?

1. Follow steps above
2.
3.

### Environment (with versions)?

Grafana: v10.4.2 (22809dea50)
OS: 10
Browser: edge

### Grafana platform?

A downloaded binary

**Comment by craftzneko** [comment]
I have found the problem but have no idea how this has happened between grafana updates. The dashboard has not been updated for over a year, but looking at the JSON at the link i see these appearing now. I suggest this ia bug that needs to be looked at why these appear, Recreating the link does NOT remove these, you need to modify the JSON directly

![image](https://github.com/grafana/grafana/assets/662108/630cb997-028f-4c30-966a-2d4eeaf22c96)

**Comment by maxcmiller** [comment]
I've had this issue too, the characters appear to be zero width spaces (`0xfeff`). Manually removing these from the JSON also fixed it for me, thanks!

<img width="914" alt="Screenshot 2024-05-15 at 09 52 35" src="https://github.com/grafana/grafana/assets/5534537/8b2f8a15-2b1f-4b4c-9129-d43ed5696c3e">

**Comment by tonypowa** [comment]
hi @craftzneko 

thank you for creating this issue

can you share the panel JSON so we can try to reproduce the issue from our side? thank you

You can copy and paste all relevant data using the panel --> more --> get help workflow in the panel's dropdown menu

**Comment by tonypowa** [comment]
Closing this issue due to inactivity. Please feel free to reopen if the problem persists.

**Comment by craftzneko** [comment]
Is the panel json going to help solve this issue? I guess the problem occurs because of some issue in the code when you add a link that then updates the panelJSON with those zero width white spaces in. I dont really want to provide the panel json as there may be confidential items (names of series etc) in there and i dont have time to go through a 1000+ lines of json to check. I solved the issue by manually removing, as did another user with the same issue did, so I suggest other users try that if they are facing the same issue, and hpefully one day this will get picked up and solved

**Comment by danield72** [comment]
It does persist.  I recently upgraded Grafana to 11.4, and -- I knowledge it could be a coincidence -- I only noticed these characters being inserted into my URL string after the upgrade.  I *had* been working on this dashboard with 11.2, but never seen those characters until after the upgrade.  From my end, I don't know what triggers it (what causes these characters to be inserted), but the following case discusses it, and it seems to have to do with dismissing the auto-fill popup:

https://github.com/grafana/grafana/issues/25107

While there is a viable *workaround* (manually remove those otherwise non-printable characters with an editor that can display them, and re-save the .json file), the permanent solution would be to detect how it occurs and stop it from getting those characters sneaking in there in the first place.  Most users wouldn't know what to do.
